src/java.base/unix/native/libmallochooks/mallochooks.c

```diff
@@ -121,6 +121,8 @@ static void* calloc_by_malloc(size_t elems, size_t size);
 static int posix_memalign_by_aligned_alloc(void** ptr, size_t align, size_t size);
 static void* memalign_by_aligned_alloc(size_t align, size_t size);
 
+#define NEEDS_MALLOC_FALLBACK
```

Comment:
How does this define make sense? Can't you just add the new code unconditionally? Or is there also some other place that could do `#undef NEEDS_MALLOC_FALLBACK` ?

Reply:
Currently it's only defined for musl-based systems.

Reply:
Ah, you're right. I should have looked where exactly the define resides... 😄